src/main/java/appeng/recipes/handlers/InscriberRecipeSerializer.java

```diff
@@ -52,7 +55,15 @@ public InscriberRecipe fromJson(ResourceLocation recipeId, JsonObject json) {
 
         InscriberProcessType mode = getMode(json);
 
-        ItemStack result = ShapedRecipe.itemStackFromJson(GsonHelper.getAsJsonObject(json, "result"));
+        JsonObject resultObj = GsonHelper.getAsJsonObject(json, "result");
+        ItemStack result = new ItemStack(ShapedRecipe.itemFromJson(resultObj));
+        if (resultObj.has("data")) {
+            try {
+                result.setTag(TagParser.parseTag(GsonHelper.getAsString(resultObj, "data")));
+            } catch (CommandSyntaxException e) {
+                throw new JsonSyntaxException("Invalid data tag:" + e);
+            }
+        }
```

Comment:
(See above, I thought there was a way to directly deserialize JSON objects into NBT without going via string)

Reply:
Not particularly, and I had to wrestle with that myself for a bit longer than I was expecting.